File: MultiLanguagePromptsFieldNote002/sales_analysis.py
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

# Generate synthetic dataset
np.random.seed(42)
dates = pd.date_range(start='2018-01-01', periods=100, freq='M')
sales = np.random.randint(100, 500, size=100) + np.arange(100) * 2 
df = pd.DataFrame({'Date': dates, 'Sales': sales})

# Calculate 3-month moving average
df['MA3'] = df['Sales'].rolling(window=3).mean()

# Visualization
sns.set_theme(style="white")
plt.figure(figsize=(12, 6))

# Scatter plot
plt.scatter(df['Date'], df['Sales'], color='#0077b6', alpha=0.6, label='Monthly Sales', s=50)

# Moving average plot
plt.plot(df['Date'], df['MA3'], color='#d62828', linewidth=2.5, label='3-Month Moving Average')

# Aesthetic styling
plt.gca().spines['top'].set_visible(False)
plt.gca().spines['right'].set_visible(False)
plt.grid(False)
plt.title('Monthly Sales Analysis', fontsize=16, fontweight='bold', color='#333333')
plt.xlabel('Date', fontsize=12)
plt.ylabel('Sales Volume', fontsize=12)
plt.legend(frameon=False)

plt.tight_layout()
plt.show()

File: MultiLanguagePromptsFieldNote002/sales_analysis.m
% Generate synthetic dataset
rng(42); 
t = (1:100)';
dates = datetime(2018, 1, 1) + calmonths(t-1);
sales = randi([100, 500], [100, 1]) + (t * 2);

% Calculate 3-month moving average
ma3 = movmean(sales, 3);

% Visualization
figure('Color', 'w');
scatter(dates, sales, 40, [0, 0.466, 0.714], 'filled', 'MarkerFaceAlpha', 0.6); hold on;
plot(dates, ma3, 'Color', [0.84, 0.157, 0.157], 'LineWidth', 2.5);

% Aesthetic styling
title('{\bf Monthly Sales Analysis}', 'FontSize', 16, 'Color', [0.2 0.2 0.2]);
xlabel('Date', 'FontSize', 12);
ylabel('Sales Volume', 'FontSize', 12);
grid off;
ax = gca;
ax.Box = 'off';
ax.TickDir = 'out';
ax.XAxis.LineWidth = 1;
ax.YAxis.LineWidth = 1;

legend('Monthly Sales', '3-Month Moving Average', 'Location', 'best', 'EdgeColor', 'none', 'Color', 'none');
hold off;
